### PR TITLE
Spotless config for kotlin

### PR DESCRIPTION
**Pull request by oldergod**
Excluding all golden files.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -39,6 +41,17 @@ allprojects {
 }
 
 subprojects {
+  apply(plugin = "com.diffplug.spotless")
+  configure<SpotlessExtension> {
+    setEnforceCheck(false)
```
Comment (oldergod) [inline build.gradle.kts:46]: So that CI doesn't break when formatting is off.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,10 @@
+root = true
```
Comment (swankjesse) [inline .editorconfig:1]: 👍🏻👍🏻

File: build.gradle.kts (hunk 1 above)
Comment (swankjesse) [reply build.gradle.kts:46]: Thanks.

File: wire-library/.editorconfig [hunk 3]
```diff
@@ -0,0 +1,10 @@
+root = true
```
Comment (swankjesse) [inline wire-library/.editorconfig:1]: Is this file necessary?

Comment (oldergod) [reply wire-library/.editorconfig:1]: Hmmm, maybe not? I thought I needed one because wire-library is its own project. I'm not sure how spotless look for editorconfig
I did a quick test locally and it seems to apply properly without this.

I'll remove for now
